hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/utils/TransactionUtils.java

```diff
@@ -78,7 +79,13 @@ public static Option<HoodieCommitMetadata> resolveWriteConflictIfAny(
       final ConcurrentOperation thisOperation = new ConcurrentOperation(currentTxnOwnerInstant.get(), thisCommitMetadata.orElse(new HoodieCommitMetadata()));
       instantStream.forEach(instant -> {
         try {
-          ConcurrentOperation otherOperation = new ConcurrentOperation(instant, table.getMetaClient());
+          ConcurrentOperation otherOperation;
+          if (resolutionStrategy.getClass().getName().equals(BucketIndexConcurrentFileWritesConflictResolutionStrategy.class.getName())) {
+            otherOperation = new ConcurrentOperation(instant, TimelineUtils.getCommitMetadata(instant,
+                reloadActiveTimeline ? table.getMetaClient().reloadActiveTimeline() : table.getActiveTimeline()));
+          } else {
```

Comment:
do we need to realod the timeline for each instant ?

Reply:
Thanks for the heads-up. It's no need to reload the timeline here.